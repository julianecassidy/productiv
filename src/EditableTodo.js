import React, { useState } from "react";
import Todo from "./Todo";
import TodoForm from "./TodoForm";

/** Show editable todo item.
 *
 * Props
 * - toDo
 * - update(): fn to call to update a todo
 * - remove(): fn to call to remove a todo
 *
 * State
 *  - beingEdited: Boolean
 *
 * EditableTodoList -> EditableTodo -> { Todo, TodoForm }
 */

function EditableTodo({ toDo, update, remove }) {
  const [beingEdited, setBeingEdited] = useState(false);
  console.log("editedableTodo, beingEditedState: ", beingEdited);

  const toDoUpdateData = {
    title: toDo.title,
    description: toDo.description,
    priority: toDo.priority,
    id: toDo.id,
  };
  // TODO: REWRITE docstring to what it's actually doing.
  /** Toggle if this is being edited */
  function toggleEdit() {
    setBeingEdited(true);
  }

  /** Call remove fn passed to this. */
  function handleDelete() {
    remove(toDo.id);
  }

  /** Edit form saved; toggle isEditing and update in ancestor. */
  function handleSave(formData) {
    update(formData);
    setBeingEdited(false);
  }

  return (
    <div className="EditableTodo">
      {beingEdited ? (
        <TodoForm initialFormData={toDoUpdateData} handleSave={handleSave} />
      ) : (
        <div className="mb-3">
          <div className="float-end text-sm-end">
            <button
              className="EditableTodo-toggle btn-link btn btn-sm"
              onClick={toggleEdit}
            >
              Edit
            </button>
            <button
              className="EditableTodo-delBtn btn-link btn btn-sm text-danger"
              onClick={handleDelete}
            >
              Del
            </button>
          </div>
          <Todo
            key={toDo.id}
            id={toDo.id}
            title={toDo.title}
            description={toDo.description}
            priority={toDo.priority}
          />
        </div>
      )}
    </div>
  );
}

export default EditableTodo;
