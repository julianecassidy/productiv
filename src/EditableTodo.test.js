import React from "react";
import { render } from "@testing-library/react";
import EditableTodo from "./EditableTodo";

const TODO = {title: "test", description: "test desc", priority: 1, id: 1}
// TODO: add a blank function and pass this in as a prop, then test if it is called

describe("EditableTodo component", function () {
  it("renders without crashing", function () {
    render(<EditableTodo />);
  });

  it("matches snapshot", function () {
    const { container } = render(<EditableTodo toDo={TODO}  />);
    expect(container).toMatchSnapshot();
  });

});