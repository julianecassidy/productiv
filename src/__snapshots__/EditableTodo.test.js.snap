// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditableTodo component matches snapshot 1`] = `
<div>
  <div
    class="EditableTodo"
  >
    <div
      class="mb-3"
    >
      <div
        class="float-end text-sm-end"
      >
        <button
          class="EditableTodo-toggle btn-link btn btn-sm"
        >
          Edit
        </button>
        <button
          class="EditableTodo-delBtn btn-link btn btn-sm text-danger"
        >
          Del
        </button>
      </div>
      <div
        class="Todo"
        id="1"
      >
        <div>
          <b>
            test todo
          </b>
           
          <small>
            Priority: 
            1
          </small>
        </div>
        <div>
          <small>
            test desc
          </small>
        </div>
      </div>
    </div>
  </div>
</div>
`;
